### PR TITLE
Unit Test -- Refactored add addressable using operator pattern and added tests.

### LINKED ISSUE #1517: Unit Test -- Core-Metadata Add Addressables
Related to https://github.com/edgexfoundry/edgex-go/projects/11#card-22883564

### PR DESCRIPTION
**Pull request by brandonforster**
Fix #1517 

Refactors the AddAddressable logic and adds unit test for that operator and executor.

Signed-off-by: Brandon Forster <brandonforster@gmail.com>

### CODE REVIEW

File: internal/core/metadata/operators/addressable/add_test.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+func TestExecutor(t *testing.T) {
+	success := SuccessfulDatabaseResult[0]
+
+	tests := []struct {
+		name             string
+		mockDb           AddressLoader
+		addr             contract.Addressable
+		expectedResult   string
+		expectedError    bool
+		expectedErrorVal error
```
Comment (jdharms) [inline internal/core/metadata/operators/addressable/add_test.go]: It looks like you're not using this field.  I'd use it to ensure you're getting the `ErrDuplicateName` you're expecting in the duplicate test case. See TestAllAddressablesExecutor  in metadata/operators/addressable/get_test.go for an example.

File: internal/core/metadata/operators/addressable/db.go [hunk 2]
```diff
@@ -5,6 +5,7 @@ import contract "github.com/edgexfoundry/go-mod-core-contracts/models"
 
 // AddressLoader provides functionality for obtaining Addressables.
 type AddressLoader interface {
+	AddAddressable(addressable contract.Addressable) (string, error)
```
Comment (jdharms) [inline internal/core/metadata/operators/addressable/db.go]: This needs to be in a different interface I believe, as it's not used "for obtaining Addressables."  I'd suggest creating an "AddressSaver" or something similar interface.

**Comment by jdharms** [comment]
recheck

File: internal/core/metadata/operators/addressable/db.go (hunk 2 above)
Comment (jdharms) [reply internal/core/metadata/operators/addressable/db.go]: Maybe `AddressAdder`

**Comment by brandonforster** [comment]
recheck

**Pull request merged**